src/xo-app/settings/ips/index.js

```diff
@@ -0,0 +1,304 @@
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import ActionRowButton from 'action-row-button'
+import BaseComponent from 'base-component'
+import DebounceInput from 'react-debounce-input'
+import includes from 'lodash/includes'
+import find from 'lodash/find'
+import findIndex from 'lodash/findIndex'
+import forEach from 'lodash/forEach'
+import Icon from 'icon'
+import isEmpty from 'lodash/isEmpty'
+import isObject from 'lodash/isObject'
+import keys from 'lodash/keys'
+import map from 'lodash/map'
+import React from 'react'
+import SortedTable from 'sorted-table'
+import store from 'store'
+import TabButton from 'tab-button'
+import { addSubscriptions } from 'utils'
+import { injectIntl } from 'react-intl'
+import { Text } from 'editable'
+import { confirm } from 'modal'
+import { Container, Row, Col } from 'grid'
+import SingleLineRow from 'single-line-row'
+import { SelectNetwork } from 'select-objects'
+import { formatIps, getNextIpV4, parseIpPattern } from 'ip'
+import { getObject } from 'selectors'
+import {
+  createIpPool,
+  deleteIpPool,
+  setIpPool,
+  subscribeIpPools
+} from 'xo'
+
+const FULL_WIDTH = { width: '100%' }
+const NETWORK_FORM_STYLE = { maxWidth: '40em' }
+
+@addSubscriptions({
+  ipPools: subscribeIpPools
+})
+@injectIntl
+export default class Ips extends BaseComponent {
+  _create = () => {
+    const { name, ips: { value: pattern }, networks } = this.refs
```

Comment:
For such a small static form, I think it's actually easier to use refs since nothing has to be checked while the user is typing. This way I don't have to manage state variables or even debounce all my inputs.